h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -1408,7 +1425,111 @@ public Submodel getSubmodel(double lambdaCVEstimate) {
     }
     return res;
   }
-  
+
+
+  // create scoring history table using scoreKeeper arrays scored_train, scored_xval, scored_valid
+  public TwoDimTable createScoringHistoryTable(boolean hasValidation,
+                                                      boolean hasCrossValidation) {
+    ModelCategory modelCategory = _output.getModelCategory();
+    boolean isClassifier = (modelCategory == ModelCategory.Binomial || modelCategory == ModelCategory.Multinomial
+            || modelCategory == ModelCategory.Ordinal);
+    List<String> colHeaders = new ArrayList<>();
+    List<String> colTypes = new ArrayList<>();
+    List<String> colFormat = new ArrayList<>();
+    colHeaders.add("Timestamp"); colTypes.add("string"); colFormat.add("%s");
+    colHeaders.add("Duration"); colTypes.add("string"); colFormat.add("%s");
+    colHeaders.add("Evaluation_Iterations"); colTypes.add("int"); colFormat.add("%d");
+    setColHeader(colHeaders, colTypes, colFormat, modelCategory, isClassifier, "Training");
+    if (hasValidation)
+      setColHeader(colHeaders, colTypes, colFormat, modelCategory, isClassifier, "Validation");
+    if (hasCrossValidation)
+      setColHeader(colHeaders, colTypes, colFormat, modelCategory, isClassifier, "Cross-Validation");
+
+    final int rows = _output._scored_train == null ? 0 : _output._scored_train.size();
+    String[] s = new String[0];
+    TwoDimTable table = new TwoDimTable(
+            "Scoring History", null,
+            new String[rows],
+            colHeaders.toArray(s),
+            colTypes.toArray(s),
+            colFormat.toArray(s),
+            "");
+    int row = 0;
+    if (null == _output._scored_train)
+      return table;
+    int iteration = 0;
+    for (ScoreKeeper si : _output._scored_train) { // fill out the table
+      int col = 0;
+      assert (row < table.getRowDim());
+      assert (col < table.getColDim());
+      DateTimeFormatter fmt = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss");
+      long training_time_ms = _output._training_time_ms.get(row);
+      table.set(row, col++, fmt.print(training_time_ms));
+      table.set(row, col++, PrettyPrint.msecs(training_time_ms-_output._start_time, true));
```

Comment:
I am pretty sure this is supposed to be a duration of this stop not its time from beggining - there are missing tests for this code